### PR TITLE
Zeo/relative object merge

### PR DESCRIPTION
**Pull request by zixiliu**
# Relative Object V2P Support, Mesh-Vertex Reward, and Agent Smoke Documentation                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                      
  ## Summary                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                      
  This PR brings a narrowed, main-ready subset of `zeo/relative-object` into `main`.                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                      
  It includes the relative-object V2P reward/command support we want to keep, mesh-vertex fine tracking reward support, curriculum/config hooks for those rewards, and                                                                                                                
  agent-facing documentation for local smoke testing. It also removes obsolete experiment examples and two-stage helper code from the committed experiment surface.                                                                                                                   
                                                                                                                                                                                                                                                                                      
  ## Included                                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                      
  ### Core Functionality                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                      
  - Added multi-object relative-pose support to `DualHandsObjectTrackingCommand`.                                                                                                                                                                                                     
  - Added relative-object metrics:                                                                                                                                                                                                                                                    
    - `relative_object_pos_error`                                                                                                                                                                                                                                                     
    - `relative_object_rot_error`                                                                                                                                                                                                                                                     
  - Added relative-object reward terms:                                                                                                                                                                                                                                               
    - `relative_object_pose_reward`                                                                                                                                                                                                                                                   
    - `relative_object_pos_reward`                                                                                                                                                                                                                                                    
    - `relative_object_rot_reward`                                                                                                                                                                                                                                                    
    - `inter_object_proximity_reward`                                                                                                                                                                                                                                                 
  - Kept `dexmachina_contact_tracking_reward`.                                                                                                                                                                                                                                        
  - Added `object_meshvert_tracking_fine` for mesh-vertex based object tracking.                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                      
  ### Curriculum and Config Support                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                      
  - Added config entries for relative-object rewards.                                                                                                                                                                                                                                 
  - Added fixed-timestep curriculum support for scheduled relative-object, DexMachina contact, and mesh-vertex reward weights.                                                                                                                                                        
  - Kept `contact_wrench_support_ratio_{right,left}` as optional metric gates.                                                                                                                                                                                                        
  - Removed unused/refinement reward terms:                                                                                                                                                                                                                                           
    - `object_keypoints_tracking_refine`                                                                                                                               
    - `object_position_tracking_fine`                                                                                                                                  
    - `object_velocity_tracking_exp`                                                                                                                                   
  - Removed `contact_bodies_coverage_frac_{right,left}` metrics and config hooks.                                                                                      
  - Removed `TerminationAnnealingCurriculum`.                                                                                                                          
                                                                                                                                                                       
  ### Experiment Runner Cleanup                                                                                                                                        
                                                                                                                                                                       
  - Removed obsolete committed experiment examples:                                                                                                                    
    - `example_osmo_sequences`                                                                                                                                         
    - `example_param_sweep`                                                                                                                                            
    - `example_sequence_list`                                                                                                                                          
    - `example_stage1_nocoll`                                                                                                                                          
    - `example_stage2`                                                                                                                                                 
    - `example_two_stage`                                                                                                                                              
  - Removed obsolete two-stage helpers:                                                                                                                                
    - `experiments/launch_stage2.py`                                                                                                                                   
    - `experiments/monitor_two_stage.py`                                                                                                                               
    - `.claude/commands/launch-two-stage.md`                                                                                                                           
  - Streamlined `experiments/README.md` around the currently committed experiment registry.                                                                            
  - Kept `example_single` as the minimal committed smoke-test experiment.                                                                                              
                                                                                                                                                                       
  ### Documentation                                                                                                                                                    
                                                                                                                                                                       
  - Added agent-facing local smoke-test docs for:                                                                                                                      
    - dummy agent                                                                                                                                                      
    - `train.py`                                                                                                                                                       
    - `eval.py`                                                                                                                                                        
  - Clarified local TensorBoard logging when W&B is not configured.                                                                                                    
  - Pointed agents to `workflow/README.md` for OSMO / NGC / cloud setup.                                                                                               
  - Updated experiment-runner docs to use the current `robotic_grounding/experiments/run_experiment.py` path.                                                          
                                                                                                                                                                       
  ## Excluded Intentionally                                                                                                                                            
                                                                                                                                                                       
  This PR intentionally does not include:                                                                                                                              
                                                                                                                                                                       
  - Meshes                                                                                                                                                             
  - URDFs                                                                                                                                                              
  - Generated assets                                                                                                                                                   
  - Large experiment outputs                                                                                                                                           
  - `motion_schema` changes                                                                                                                                            
  - `scene_utils` changes                                                                                                                                              
  - OakInk2/ManipTrans/SPIDER baseline plumbing                                                                                                                        
  - Obsolete multi-run example experiment directories                                                                                                                  
  - Two-stage launch/monitor pipeline code                                        
  
  
   ## Validation

  Performed on this branch:

  - Rebased onto `origin/main`.
  - Verified PR scope against `origin/main`.
  - `git diff --check` passed.
  - Focused Python syntax checks passed with `py_compile`.
  - Focused `pre-commit` from `robotic_grounding/` passed:
    - ruff
    - black
    - mypy
    - python ast
    - trailing whitespace / EOF checks
  - Local dummy-agent GUI smoke test passed.
  - Local `train.py` GUI smoke test passed with:
    - `Sharpa-V2P-v0`
    - `arctic/arctic_processed/arctic_s01_box_grab_01/sharpa_wave`
    - `num_envs=1`
    - `max_iterations=1`
    - TensorBoard logger
  - Local eval smoke command prepared against:
    - `logs/rsl_rl/sharpa_v2p/2026-06-11_14-24-13_smoke_train/model_0.pt`

  ## Current Branch Commit Breakdown

  - `5fd69717` — Add relative-object V2P merge payload
  - `c45ba33b` — Register relative-object example experiments
  - `36751db5` — Document agent smoke tests and experiment workflow
  - `c427ede1` — Prune relative object merge scope
  - `66515873` — Streamline experiments README
  - `6e4e82cb` — Remove stage2 launch helper
  - `b2751214` — Remove termination annealing curriculum
  - `4c30afcd` — Drop scene utils changes
  - `31ecee2f` — Remove unused object fine rewards
  - `08b3e6f9` — Remove contact coverage metrics

  If eval did complete successfully, change the last validation bullet to:

  - Local `eval.py` GUI smoke test passed against:
    - `logs/rsl_rl/sharpa_v2p/2026-06-11_14-24-13_smoke_train/model_0.pt`

### CODE REVIEW

File: robotic_grounding/experiments/README.md [hunk 1]
```diff
@@ -29,6 +31,32 @@ Add your own experiment folders locally; they are gitignored (`experiments/exp*/
...
 
+### Relative-object merge examples
+
+These are the committed examples to use when validating the relative-object merge path:
+
+| ID | Purpose |
+|----|---------|
+| `example_fixed_post` | Stage 2 FixedTimestepCurriculum with the stage-3 object-tracking boost in the tail. |
+| `example_AC_post` | Adaptive custom VOC schedule with the stage-3 object-tracking boost in the same run. |
+| `example_pre_fixed_post` | Pipeline wrapper that runs collision-free stage 1, then fixed-post stage 2/3. |
```
Comment (huihuaNvidia2023) [inline robotic_grounding/experiments/README.md]: there are also tasks like `example_pre_fixed_post_stage1` and `...._stage2`. Should we remove those or added to the table?

File: robotic_grounding/source/robotic_grounding/robotic_grounding/motion_schema/motion_schema.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
+# MotionData: Unified Motion Parquet Schema
```
Comment (huihuaNvidia2023) [inline robotic_grounding/source/robotic_grounding/robotic_grounding/motion_schema/motion_schema.md]: this file has been removed. Please rebase the tip of the tree.

File: robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/v2p/v2p_hand_env_cfg.py [hunk 3]
```diff
@@ -430,6 +543,33 @@ class TerminationsCfg:
...
+        func=mdp.hand_object_contact_violation,
+        params={
+            "command_name": "dual_hands_object_tracking_command",
+            "grasp_distance_threshold": 0.0,
+            "contact_force_threshold_n": 0.5,
+            "grace_frames": 5,
+        },
+    )
+
+    # Disabled: this per-finger Cartesian-error termination no longer serves
```
Comment (huihuaNvidia2023) [inline robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/v2p/v2p_hand_env_cfg.py]: should we just remove this?

File: robotic_grounding/experiments/example_AC_post/config.yaml
Comment (xinghaozhu-nvidia) [inline robotic_grounding/experiments/example_AC_post/config.yaml:1]: Could we clean the experiments folder to make it clearer?

File: robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/scene_utils/scene_config.py [hunk 4]
```diff
@@ -16,6 +16,92 @@
...
+    }
+
+
+# Per-object scale overrides for OakInk-V2 objects whose meshes are repaired as
+# solid blobs and need a small uniform scale tweak to avoid PhysX interpenetration.
+# Ported verbatim from ManipTrans (main/dataset/oakink2_dataset_utils.py:34-41).
+# Keys use OakInk-V2's @ notation; lookup also accepts the body_name "safe" form
+# (underscores) used downstream. Effect: shrink the body, grow the cap, opening
+# ~10mm of radial clearance — enough for the cap to wrap the body's neck.
+OAKINK2_OBJECT_SCALE: dict[str, float] = {
```
Comment (xinghaozhu-nvidia) [inline robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/scene_utils/scene_config.py]: Can we move these magic numbers to a standalone file?

File: robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/v2p/mdp/commands/hand_object_commands.py [hunk 5]
```diff
@@ -1888,184 +2174,164 @@ def _compute_contact_wrench_supports(self, side: str) -> None:
...
         if not self.cfg.enable_additional_metrics:
             return
 
-        voc_scale_now = float(self.virtual_object_controller_scale_factor)
-        decay_fired = 1.0 if voc_scale_now < self._prev_voc_scale else 0.0
-        self._prev_voc_scale = voc_scale_now
-        self.metrics["voc_decay_fired"] = decay_fired * torch.ones(
-            self.num_envs, device=self.device
-        )
+        # DISABLED: voc_decay_fired W&B logging
```
Comment (xinghaozhu-nvidia) [inline robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/v2p/mdp/commands/hand_object_commands.py]: A lot of commented code, we need to get rid of it

File: robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/scene_utils/apply_scene_config.py [hunk 6]
```diff
@@ -132,6 +150,118 @@ def _spawn_rigid(
...
+            attr_name
+        )
+    # Enable scene-wide CCD so PhysX does swept time-of-impact checks for fast-
+    # moving bodies (hand fingertips). Door is 3.5cm thick; without CCD even
+    # 8cm contact_offset can miss tunneling at 100Hz with fast wrist motion.
+    if hasattr(env_cfg, "sim") and hasattr(env_cfg.sim, "physx"):
+        env_cfg.sim.physx.enable_ccd = True
+
+
+def _apply_maniptrans_oakink_spawn_overrides(
```
Comment (xinghaozhu-nvidia) [inline robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/scene_utils/apply_scene_config.py]: We don't need baseline methods in the main branch; otherwise, we need to maintain them.

File: robotic_grounding/experiments/example_AC_post/config.yaml
Comment (zixiliu) [reply robotic_grounding/experiments/example_AC_post/config.yaml:1]: Done

File: robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/v2p/mdp/commands/hand_object_commands.py (hunk 5 above)
Comment (zixiliu) [reply robotic_grounding/source/robotic_grounding/robotic_grounding/tasks/v2p/mdp/commands/hand_object_commands.py]: done

File: robotic_grounding/experiments/README.md (hunk 1 above)
Comment (zixiliu) [reply robotic_grounding/experiments/README.md]: Removed and updated

**Review by huihuaNvidia2023 — APPROVED** [review]
LGTM

**Comment by zixiliu** [comment]
After streamlining, verified that dummy agent, train.py, eval.py still run locally in docker container.
